libraries/rust/src/events/market.rs

```diff
@@ -0,0 +1,278 @@
+use apca::api::v2::clock::Get;
+use apca::{ApiInfo, Client};
+use chrono::{DateTime, Utc};
+use cloudevents::{Event, EventBuilder, EventBuilderV10};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use tracing::error;
+use uuid::Uuid;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum ClosedReason {
+    AfterHours,
+    BeforeHours,
+    Weekend,
+    Holiday,
+    Unknown,
+}
+
+impl ToString for ClosedReason {
+    fn to_string(&self) -> String {
+        match self {
+            ClosedReason::AfterHours => "after_hours".to_string(),
+            ClosedReason::BeforeHours => "before_hours".to_string(),
+            ClosedReason::Weekend => "weekend".to_string(),
+            ClosedReason::Holiday => "holiday".to_string(),
+            ClosedReason::Unknown => "unknown".to_string(),
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum Status {
+    Open,
+    ExtendedHours,
+    Closed(ClosedReason),
+    Unknown,
+}
+
+impl ToString for Status {
+    fn to_string(&self) -> String {
+        match self {
+            Status::Open => "open".to_string(),
+            Status::ExtendedHours => "extended_hours".to_string(),
+            Status::Closed(reason) => format!("closed_{}", reason.to_string()),
+            Status::Unknown => "unknown".to_string(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Market {
+    client: Client,
+    /// The time the market status was last checked.
+    pub updated_at: Option<DateTime<Utc>>,
+    /// The current market status.
+    pub status: Status,
+    /// The current market timestamp.
+    pub current: Option<DateTime<Utc>>,
+    /// The next market open timestamp.
+    pub next_open: Option<DateTime<Utc>>,
+    /// The next market close timestamp.
+    pub next_close: Option<DateTime<Utc>>,
```

Comment:
We don't need these in-line comments.

Reply:
these are for the doc sitt, 3 `/` comments will render as a docstring

Reply:
Gotcha. I'm down to experiment with them for now but I suspect some might not be needed if the struct is self-explanatory enough.